File: kotlin-spring/src/main/java/com/example/kotlinspring/KotlinSpringApplication.java
package com.example.kotlinspring;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.*;

@SpringBootApplication
@CrossOrigin(origins = "*")
@RestController
public class KotlinSpringApplication {

    public static void main(String[] args) {
        SpringApplication.run(KotlinSpringApplication.class, args);
    }

     int Fibonacci(int n) {
        if (n <= 1) {
            return n;
        }
        return Fibonacci(n - 1) + Fibonacci(n - 2);
    }

    @GetMapping(value = "/healthcheck", produces = "application/json")
    public ResponseEntity<String> healthcheck() {
        return new ResponseEntity<>("\"Ok\"", HttpStatus.OK);
    }

    @GetMapping(value = "/employees", produces = "application/json")
    public ResponseEntity<List<Employee>> getEmployees() {
        List<Employee> employees = new ArrayList<>();
        employees.add(new Employee(1, "John", "Doe"));
        employees.add(new Employee(2, "Peter", "Parker"));
        employees.add(new Employee(3, "Tony", "Stark"));
        return new ResponseEntity<>(employees, HttpStatus.OK);
    }

    @PostMapping(value = "/employees", produces = "application/json")
    public ResponseEntity<Integer> createEmployee() {
        int n = 40;
        int fib = Fibonacci(n);
        return new ResponseEntity<>(fib, HttpStatus.OK);
    }

}

File: kotlin-spring/src/main/kotlin/com/example/kotlinspring/KotlinSpringApplication.kt
package com.example.kotlinspring

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.CrossOrigin
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RestController

@SpringBootApplication
@CrossOrigin(origins = ["*"])
class KotlinSpringApplication

fun main(args: Array<String>) {
    runApplication<KotlinSpringApplication>(*args)
}

@RestController
class ExampleResource {
    @GetMapping(value = ["/healthcheck"], produces = ["application/json"])
    fun doHealthcheck():ResponseEntity<String> = ResponseEntity("\"Ok\"", HttpStatus.OK)

    @GetMapping("/employees")
    fun getEmployees(): ResponseEntity<List<Employee>> = ResponseEntity(listOf(
        Employee(1, "John", "Doe"),
        Employee(2, "Peter", "Parker"),
        Employee(3, "Tony", "Stark")),
        HttpStatus.OK
    )

    @PostMapping("/employees")
    fun postEmployees(): ResponseEntity<Int> {
        Thread.sleep(300)
        return ResponseEntity(42, HttpStatus.OK)
    }
}

data class Employee(val id: Int, val firstname: String, val lastname: String)
